File: recursion/factorial.js
const factorial = (n) => {
  if (n === 2) {
    return 2
  } else {
    return factorial(n-1)*n
  }
}

console.log(factorial(5))

const factorialIterative = (n) => {
  let factorial = n
  while (n > 2) {
    factorial = factorial * (n-1)
    n--;
  }
  return factorial
}

console.log(factorialIterative(5));

File: recursion/factorial.py
def factorialRecursive(n):
    if n == 2:
        return 2
    else:
        return factorialRecursive(n-1) * n

print(factorialRecursive(5))

def factorialIterative(n):
    factorial = n
    while n > 2:
        factorial = factorial * (n-1)
        n -= 1
    return factorial

print(factorialIterative(5))
